### PR TITLE
HTCONDOR-1426-fix-curl-segfault

### PR DESCRIPTION
**Pull request by GregThain**
https://opensciencegrid.atlassian.net/browse/HTCONDOR-1426

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_tests/test_curl_plugin.py [hunk 1]
```diff
@@ -85,13 +91,28 @@ def job_with_good_url(default_condor, good_url, test_dir, path_to_sleep):
...
+def job_with_good_file_url(default_condor, good_file_url, test_dir, path_to_sleep):
+    job = default_condor.submit(
+        {
+            "executable": path_to_sleep,
+            "arguments": "0",
+            "log": (test_dir / "good_file_url.log").as_posix(),
+            "transfer_input_files": good_file_url,
+            "transfer_output_files": "file_url_tester",
+            "should_transfer_files": "YES",
+            "requirements": "(SlotID == 1)"
```
Comment (Todd-L-Miller) [inline src/condor_tests/test_curl_plugin.py:104]: Why only slot 1?

**Review by Todd-L-Miller — COMMENTED** [review]
Good catch about missing the plugin exiting on a signal.

File: src/condor_tests/test_curl_plugin.py (hunk 1 above)
Comment (GregThain) [reply src/condor_tests/test_curl_plugin.py:104]: Because the other tests that test successful file xfer use this slot.

**Comment by Todd-L-Miller** [comment]
I also liked the submitting-it-with-the-failing test thing, to make sure that the test didn't incorrectly succeed on the weird platforms that were actually having the problem.

**Review by Todd-L-Miller — APPROVED** [review]
Let the tests finish, just to be safe, but otherwise merge when ready.

**Comment by GregThain** [comment]
Tests passed, except for a couple of build failures where the platforms could not pull from their home repos.  We'll call this good.

**Pull request merged**